src/v/cluster/members_manager.cc

```diff
@@ -841,6 +841,30 @@ model::node_id members_manager::get_node_id(const model::node_uuid& node_uuid) {
 }
 
 ss::future<> members_manager::set_initial_state(
```

Comment:
> cluster/members_manager: make sure set_initial_state is atomic
> eliminate a race condition between bootstrap and termination

Please state in the commit message what the race condition is, how it happens, and how the commit fixes it.